crates/forge/src/lib.rs

```diff
@@ -462,26 +518,63 @@ fn run_with_fuzzing(
     };
     let mut fuzzer = RandomFuzzer::create(fuzzer_seed, fuzzer_runs, &args)?;
 
-    let mut results = vec![];
+    let mut tasks = vec![];
 
     for _ in 1..=fuzzer_runs {
         let args = fuzzer.next_args();
+        tasks.push(task::spawn({
+            let case = case.clone();
+            let package_root =  package_root.clone();
+            let runner = runner.clone();
+            let args = args.clone();
+            let cancellation_token = cancellation_token.clone();
+            let cancellation_fuzzing_token = Arc::new(token.clone());
+            let runner_params = runner_params.clone();
+            let runner_config = runner_config.clone();
+            async move {
+                tokio::select! {
+                    _ = cancellation_token.cancelled() => {
+                        // The token was cancelled
+                        Ok(TestCaseSummary::Failed { name: "fuzzing".to_string(), run_result: None, msg: None, arguments: vec![] })
+                    },
+                    _ = cancellation_fuzzing_token.cancelled() => {
+                        Ok(TestCaseSummary::Failed { name: "fuzzing".to_string(), run_result: None, msg: None, arguments: vec![] })
```

Comment:
Why do we need separate tokens for these?

Reply:
cancellation_token -  is to stop threads in case of `exit-first` so it stops all tests in package
cancellation_fuzzing_token - it cancels all threads related to a single fuzzing test. I assume, that for fuzzing test if one generated test fails, we can assume that it failed at all

Reply:
Okay, makes sense